cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -152,6 +155,74 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
+	//change an assignment's name
+	@PutMapping("/gradebook/{id}/{newName}")
+	@Transactional
+	public void updateAssignmentName (@PathVariable("id") Integer assignmentId, @PathVariable("newName") String newName ) {
+		
+		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
+		Assignment assignment = checkAssignment(assignmentId, email);  // check that user name matches instructor email of the course.
+		
+		assignment.setName(newName);
+		assignmentRepository.save(assignment);
```

Comment:
As long as this assignment object has the correct assignmentId, which it looks like it should - this should correctly update the existing entry. This looks good to me.